projects/laji/src/app/shared-modules/observation-map/observation-map/observation-map.component.ts

```diff
@@ -275,12 +306,14 @@ export class ObservationMapComponent implements OnChanges, OnDestroy {
       this.boxDataOptionsCache.reset();
       this.updateMap();
     } else {
-      const dataOptions = this.mapData[0];
+      const dataOptions = this.lajiMap.map.getData()[1];
       if (!dataOptions) { return; }
       const vis = this.visualization?.[this.visualizationMode];
-      if (vis?.getFeatureStyle) { dataOptions.getFeatureStyle = vis.getFeatureStyle; }
-      // changing mapData object reference so that laji-map recognizes that it has changed
-      this.mapData = [...this.mapData];
+      if (vis?.getFeatureStyle) {
+        dataOptions.getFeatureStyle = getFeatureStyleWithHoverAndActive(vis.getFeatureStyle);
+      }
+      // changing mapData object reference so that laji-map recognizes that it has changed.
+      this.mapData = this.lajiMap.map.getData();
```

Comment:
The data at index `0` was the drawn box, index `1` hold the occurrence data. I'm not sure why this worked before, as the wrong `getFeatureStyle` was being updated.

I had to change the `mapData` reference to use the `getData()` from laji-map, or the data's `activeIdx` would been wrong (the getter return the data with the current value of the `activeIdx`, using the reference to laji-map data item doesn't since laji-map doesn't mutate the given data).

Reply:
Good catch, the order of the data was changed in https://github.com/luomus/laji/commit/d3ebac5ac90549838a3fc385d3a15d961c531ce7

Reply:
```
// changing mapData object reference so that laji-map recognizes that it has changed.
this.mapData = this.lajiMap.map.getData();
```
Now this line only implicitly changes the reference (it relies on `this.lajiMap.map.getData()` returning a new reference). It seems like it works so I guess that getData function is indeed returning a new reference (?), but the comment should be changed to reflect this

Reply:
I made it update the data directly to laji-map, instead of updating the local `mapData`, that is used only for passing new data to the template. I also tightened the occurrence map visualization typings, since it always has `getFeatureStyle`.